go/vt/vttablet/tabletserver/vstreamer/vstreamer_flaky_test.go

```diff
@@ -291,6 +291,52 @@ func TestSetStatement(t *testing.T) {
 	runCases(t, nil, testcases, "current", nil)
 }
 
+// TestSetForeignKeyCheck confirms that the binlog RowEvent flags are set correctly when foreign_key_checks are on and off.
+func TestSetForeignKeyCheck(t *testing.T) {
+	testRowEventFlags = true
+	defer func() { testRowEventFlags = false }()
+
+	execStatements(t, []string{
+		"create table t1(id int, val binary(4), primary key(id))",
+	})
+	defer execStatements(t, []string{
+		"drop table t1",
+	})
+	engine.se.Reload(context.Background())
+	queries := []string{
+		"begin",
+		"insert into t1 values (1, 'aaa')",
+		"set @@session.foreign_key_checks=1",
```

Comment:
Note that this is not the same statement we're using in the new code. This is related to previous comments about using =[true|false] rather than 1|0. Better to standardize on 0|1 everywhere IMO.